docs/source/branch.rst

```diff
@@ -0,0 +1,218 @@
+
+.. code-block:: yaml
+
+    input_draw_count: 10
+
+Stochastic Uncertainty
+^^^^^^^^^^^^^^^^^^^^^^
+Vivarium simulations are probabilistic in nature. They tend to contain distributions that describe quantities rather
+than singular values. Because of this, our models are subject to stochastic uncertainty. To attempt to capture some of
+this uncertainty we can run the simulation with different random seeds which will result in different numbers drawn from
+the underlying pseudo-random number generator, and thus different output results. This can be specified in the branches
+file with the ``random_seed_count`` key. This key specifies an integer that represents the number of different random
+seeds to use, each generated randomly and run in a separate simulation.
+
+.. note::
+    Random seeds are a convenient way to scale up a simulation's population in parallel. For example, running a
+    simulation with one million simulants and a single random seed is equivalent to running the same simulation with
+    ten thousand people and 100 random seeds. However, because simulations specified with different seeds will be run
+    in parallel (link psimulate), this is often preferable.
+
+An example of specifying ``random_seed_count`` is below.
+
+.. code-block:: yaml
+
+    random_seed_count: 100
+
+Combining Draws and Seeds
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Since specifying either draws or seeds will result in multiple simulations being run, it is important to understand how
+simulation configurations are determined when both are specified. The specification of multiple branch configurations
+that would result in more than one simulation will lead to simulations for the cartesian product of the the
+configuration sets. An example may make this clearer, so consider the following model specification.
+
+.. code-block:: yaml
+
+    input_draw_count: 100
+    random_seed_count: 10
+
+It combines the two configuration keys we just learned about. Taken separately, the ``input_draw_count`` mapping would
+lead to 100 simulations on 100 random draws of input data while the ``random_seed_count`` mapping would lead to ten
+simulations on ten randomly generated random seeds. With both specified, the result is one thousand total simulations,
+one for each member of the cartesian product of those sets. That is, ten simulations with the ten random seeds for each
+of the 100 GBD draws.
+
+Parameter Variations
+--------------------
+A major function of branches files is to enable easy manipulation of the configuration parameters of a model
+specification to generate different scenarios or examine the sensitivity of a model to changes in a specific parameter.
+In the following sections we will describe a number of ways you can construct different scenarios and the effect on
+the number of simulations these will have.
+
+.. note::
+    The following examples that alter configuration parameters all lie under a ``branches`` key.
```

Comment:
this note is a little cryptic. Maybe expand a little bit? I think the point you're trying to make is that these changes are under the branches key as opposed to the input draws/seeds which are top level but it's a little unclear whether this note is just saying 'notice the branches key - it's a convenience for the following examples' or is saying 'notice that you have to specify everything except draws & seeds under a 'branches' key. Maybe move the sentence on line 113 about how the branches block changes model spec up to this note?

I also find it a little confusing that we have 'branches key', 'branches file', and 'branches configurations' used in multiple places in this doc. Maybe a clarification somewhere about what these three terms mean?

Reply:
yeah I debated this one. I just wanted to call attention to the difference to try and avoid simple syntax mistakes but maybe it is better left alone

Reply:
I agree the terminology is all over. what is a configuration, what is a parameter, does a branch include draws and seeds? I'll try and clarify